### DISCUSSION TITLE
Getting started with stac-fastapi

### DISCUSSION THREAD

**Discussion by jacobtodd91** [stac-utils/stac-fastapi#641] [Q&A]
I am brandnew at a working with STAC and have came across this repository which seems to include all the needed functionality for a project I am working on. My issue is that I am honestly not sure how to get started. I am familiar with FastAPI, PostgreSQL, and how to connect and interact with the database with both psycopg and sqlalchemy, but just cannot figure out how to get stac-fastapi up and running. Could someone provide a working main.py script that includes the connection to the database and at least one endpoint for requesting data as well as one for inserting the data? This would be extremely appreciated.

**Comment by jonhealy1** [comment]
We have an example [here](https://github.com/stac-utils/stac-fastapi-elasticsearch-opensearch/tree/main/examples) if you are wanting to use Docker. There is also a postman collection that shows some route examples. This is for `stac-fastapi.elasticsearch` but `stac-fastapi.pgstac` is similar and you can use the Dockerfile and docker-compose.yml from their repo instead. 

Let me know if this is confusing. One thing I guess is that there's no real point to run this repo on its own. You normally want to run this code via a backend like stac-fastapi-elasticsearch-opensearch or stac-fastapi-pgstac.
